File: Exercism/dart/word-count/lib/word_count.dart
class WordCount {
  Map<String, int> countWords(String input) {
    Map<String, int> result = {};

    RegExp(r"((?:\w|['-]\w)+)").allMatches(input)
    .forEach((Match it) {
        String word = it.group(0).toLowerCase();
        if (word.startsWith("'"))
            word = word.substring(1);

        if (result.containsKey(word))
            result[word] = result[word]+1;
        else
          result[word] = 1;
    });

    return result;
  }
}

File: Exercism/python/word-count/word_count.py
import re

def count_words(sentence):
    result = {}

    pattern = re.compile(r"((?:\w|['-]\w)+)")
    for word in re.findall(pattern, sentence.lower().replace("_", " ")):
        word = word[1:] if word.startswith("'") else word
        result[word] = result[word] + 1 if word in result else 1

    return result
